cdap-ui/app/cdap/components/FieldLevelLineage/v2/LineageSummary/index.tsx

```diff
@@ -101,27 +126,54 @@ class LineageSummary extends React.Component<{ classes }> {
       .attr('width', anchorHeight)
       .attr('height', anchorHeight)
       .attr('rx', anchorRx)
-      .style('fill', grey[300]);
+      .attr('pointer-events', 'fill') // To make rect clickable
+      .style('fill', edgeColor);
   }
 
-  private drawLinks() {
+  private clearCanvas() {
     // clear any existing links and anchors
     d3.select('#links-container')
       .selectAll('path,rect')
       .remove();
+  }
+
+  // Draws only active links
+  private drawActiveLinks(activeLinks) {
+    this.clearCanvas();
 
-    this.activeLinks.forEach((link) => {
-      this.drawLineFromLink(link);
+    activeLinks.forEach((link) => {
+      this.drawLineFromLink(link, true);
     });
   }
 
+  private drawLinks(links, activeFieldId = null) {
```

Comment:
Same here. Can define the types for inputs? Sorry I am repeating myself over but now that we are using typescript lets try to be diligent about defining types for a more defensive approach on UI development. I myself am not doing it fully today but let us try.

Reply:
Thanks for the reminder! I've added types here for the inputs and will try to be consistent about doing so in the future.